api/handler/object_list.go

```diff
@@ -121,32 +121,39 @@ func parseListObjectsArgsV2(r *http.Request) (*layer.ListObjectsParamsV2, error)
 	}
 	res.ListObjectsParamsCommon = *common
 
-	res.ContinuationToken = r.URL.Query().Get("continuation-token")
-	res.StartAfter = r.URL.Query().Get("start-after")
-	res.FetchOwner, _ = strconv.ParseBool(r.URL.Query().Get("fetch-owner"))
+	if val, ok := queryValues["continuation-token"]; ok {
+		if err := object.NewID().Parse(val[0]); err != nil {
+			return nil, api.GetAPIError(api.ErrIncorrectContinuationToken)
+		}
+		res.ContinuationToken = val[0]
+	}
+
+	res.StartAfter = queryValues.Get("start-after")
+	res.FetchOwner, _ = strconv.ParseBool(queryValues.Get("fetch-owner"))
 	return &res, nil
 }
 
 func parseListObjectArgs(r *http.Request) (*layer.ListObjectsParamsCommon, error) {
 	var (
-		err error
-		res layer.ListObjectsParamsCommon
+		err         error
+		res         layer.ListObjectsParamsCommon
+		queryValues = r.URL.Query()
```

Comment:
I'm not sure if we need to use it here unless we are interested in having a key in the queries (The same for `parseListObjectVersionsRequest`).

Reply:
But in `parseListObjectsArgs` code looks like:
```
...
		queryValues = r.URL.Query()
	)

	if info := api.GetReqInfo(r.Context()); info != nil {
		res.Bucket = info.BucketName
	}

	res.Delimiter = queryValues.Get("delimiter")
	res.Encode = queryValues.Get("encoding-type")
...
```
`api.GetReqInfo` doesn't throw errors and then in any case we will read keys from queryValues. I'm not sure it's really important to move `queryValues = r.URL.Query()` down. 

In `parseListObjectVersionsRequest` code looks like:
```
...	
queryValues = r.URL.Query()
	)

	if queryValues.Get("max-keys") == "" {
		res.MaxKeys = maxObjectList
...
```

We read keys from queryValues right after parsing.